connect/runtime/src/main/java/org/apache/kafka/connect/runtime/standalone/StandaloneHerder.java

```diff
@@ -220,6 +218,11 @@ private String startConnector(Map<String, String> connectorProps) {
             state.configOriginals = connectorProps;
             state.config = connConfig;
         }
+
+        if (tempConnectors.containsKey(connName)) {
```

Comment:
Right, that's what it does, but why do we need to remove them from the cache? Is there state that needs to be cleared out? If so, wouldn't we not even want to cache it since completely different validation requests would use the same connector instance?